src/queries/BaseQuery.ts

```diff
@@ -0,0 +1,53 @@
+import { PaginationParams, getPaginationParams } from '@/lib/pagination';
+
+export interface QueryParams {
+  databaseModel: any;
+  paginationParams?: PaginationParams;
+  isPaginated?: boolean;
+}
+
+interface BaseQueryInterface<T> extends QueryParams {
+  paginationQuery: { [index: string]: any };
+  filterQuery: { [index: string]: any };
+
+  query: () => Promise<T[]>;
+  call: () => Promise<T[]>;
+  buildPaginationQueryParams: () => void;
+}
+
+class BaseQuery<T> implements BaseQueryInterface<T> {
+  databaseModel: any;
+  paginationParams?: PaginationParams;
+  isPaginated: boolean;
+  paginationQuery: { [index: string]: any };
+  filterQuery: { [index: string]: any };
+
+  constructor(queryParams: QueryParams) {
+    this.databaseModel = queryParams.databaseModel;
+    this.paginationParams = queryParams.paginationParams;
+    this.isPaginated = queryParams.isPaginated ?? true;
+    this.paginationQuery = {};
+    this.filterQuery = {};
+  }
+
+  async call() {
```

Comment:
This is more idiomatic for Ruby code (it is related to lambda) but less so in a JS codebase. In addition, having two async methods `call1 and `query` seems redundant. Both could be merged together.

```suggestion
  async query() {
    this.buildPaginationQueryParams();
    
    return this.databaseModel.findMany({
      where: { ...this.filterQuery },
      ...this.paginationQuery,
    });
  }
```